### ISSUE TITLE
Screen reader not reading legends

### ISSUE THREAD

**Issue by tjn-financial-com** [highcharts/highcharts#20990]
#### Expected behaviour
Once a legend item is focused, users can use the Arrow keys (Up, Down, Left, Right) to navigate between different legend items and the screen reader should read out the corresponding legend.

#### Actual behaviour
Able to navigate through the legends using arrow keys, but the screen reader is not reading out the corresponding legend.

#### Live demo with steps to reproduce
https://jsfiddle.net/BlackLabel/8gse7v0k/show
Steps to reproduce:
- Use `TAB` key to navigate to the first legend(screen reader doesn't read out the legend)
- Use arrow keys to navigate to next legend(still not reading/announcing any of the legends)

#### Product version
Highcharts: v11.4.1

#### Affected browser(s)
Chrome

#### Screen reader used
JAWS, NVDA

**Comment by hubertkozik** [comment]
Hi @tjn-financial-com!

We are unable to reproduce your issue. Can you provide more details about the browser and screen reader versions?

**Comment by tjn-financial-com** [comment]
Hi @hubertkozik !

I've included the version details below.
JAWS: 2023.2402.1
Chrome: 124.0.6367.61
OS: Windows

Thanks!

**Comment by anil1687** [comment]
@hubertkozik Issue is reproducible, facing same issue, screen reader reading in 11.2 version but after updating to 11.3/11.4 screen reader not reading the legends and it's toggle information.

**Comment by patrickhlauke** [comment]
Issue is indeed reproducible.

Video below using Chrome/NVDA: tabbing through the first chart on https://www.highcharts.com/docs/accessibility/accessibility-module note how once the "NVDA" legend control receives focus, NVDA is completely silent. Incidentally, the `<button>` element for the legend is set to `tabindex="-1"`, and it's not what actually receives focus it seems (I suspect the *real* focus is landing on an element inside the `<svg>` that has been set to `aria-hidden="true"`, so focus is landing on "nothing" as far as the screen readers are concerned).

Uploading highcharts-legends.mp4…

The problem can also be reproduced in Safari/macOS/VoiceOver.

**Comment by marvin19** [comment]
Thanks for providing screen reader details, browser details and video @tjn-financial-com and @patrickhlauke . I was able to reproduce the issue. It looks like focus is set to the `<li>`-element that wraps the `<button>`-element, when it should be set to the `<button>`-element itself.  I will work on a fix.

_Internal note:_
Looks like it broke here: https://github.com/highcharts/highcharts/pull/19640

**Comment by patrickhlauke** [comment]
perfect, thanks @marvin19

**Comment by patrickhlauke** [comment]
I'll add here that once the actual `<button>` receives focus, it will still not be possible out of the box to then use cursor keys left/right to move between those buttons when a windows screen reader is running, as the cursor key events will be swallowed by the SR for its own navigation (since by default button doesn't "need" cursor keys, so the keystrokes are not passed on to the JS of the page). this may require further work, such coding the whole legend as an ARIA "toolbar" https://www.w3.org/WAI/ARIA/apg/patterns/toolbar/ with all the related `role` information, which then signals to the screen reader that cursor left/right keystrokes should be handed over to the page and that the JS will handle the roving `tabindex` between controls etc)

**Comment by patrickhlauke** [comment]
@TorsteinHonsi just checking - will the fix make its way to the examples in documentation like https://www.highcharts.com/docs/accessibility/accessibility-module ?

**Comment by TorsteinHonsi** [comment]
@marvin19 ☝️

**Comment by patrickhlauke** [comment]
@marvin19 great stuff, thank you
